src/cli.rs

```diff
@@ -0,0 +1,34 @@
+use clap::{Parser, Subcommand};
+
+use crate::data::Provider;
+
+pub mod prelude {
+    pub use clap::Parser;
+}
+
+/// Simple weather CLI.
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+pub struct Cli {
```

Comment:
Suggestion: Add a `--config` option for letting the user specify a custom config/storage location.  This can be used in the tests in place of `CONFIG_NAME`, which would let you get rid of the debug-only implementation of `config_name()` (It's almost never a good idea to change the code under test when testing).

When implementing this, be sure to use the `tempfile` crate to generate a location for the test config file; this way, the test configs will be automatically cleaned up.  At the moment, running the tests creates files with random names in the weather-cli config directory that persist after the tests are done.

Reply:
Really great idea, thank you! Changed in https://github.com/mhnap/weather-cli/commit/b0b05e965a39a5b09d713dcbd843b02a7553f361